src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosCoinviewRule.cs

```diff
@@ -0,0 +1,221 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Proof of stake override for the coinview rules - BIP68, MaxSigOps and BlockReward checks.
+    /// </summary>
+    [ExecutionRule]
+    public class PosCoinviewRule : PowCoinviewRule
+    {
+        /// <summary>Provides functionality for checking validity of PoS blocks.</summary>
+        public IStakeValidator StakeValidator { get; private set; }
```

Comment:
I still think this guy should be private

Reply:
This is available on the DI so private/public does not matter much.

Reply:
Yep, agreed

Reply:
I guess you never really saw spaghetti code.